test/functional/TestUtilities/src/org/openj9/test/util/StringUtilities.java

```diff
@@ -38,10 +39,19 @@
 	 * @return Optional which is either empty or contains matching string 
 	 */
 	public static Optional<String> searchSubstring(String needle, List<String> haystack) {
-		return haystack.stream().filter(s -> s.contains(needle)).findFirst();
+		return searchSubstring(needle, haystack.stream());
+	}
+	
+	/**
+	 * Find the first occurrence of a substring in a list of Strings.
+	 * @param needle pattern for which to search
+	 * @param haystack list of Strings to search
+	 * @return Optional which is either empty or contains matching string 
+	 */
+	public static Optional<String> searchSubstring(String needle, Stream<String> haystack) {
```

Comment:
Is this new method actually used (other than immediately above)? If you generalize the first method so the haystack has type `Collection<String>`, it becomes more useful, perhaps to the point where this method is no longer needed.

Reply:
> Is this new method actually used (other than immediately above)? 

Yes, it is used in `testDumps()` to scan a stream of input lines.

> If you generalize the first method so the haystack has type `Collection<String>`, it becomes more useful

Good idea.